packages/app-builder/src/components/hooks/use-introspection/nested-fields.ts

```diff
@@ -0,0 +1,58 @@
+import { IntrospectionOutputTypeRef, IntrospectionObjectType } from 'graphql'
+import { flatKind, getObjectType } from './helpers'
+import { LIST, NON_NULL, OBJECT, QueryableField, SCALAR } from './types'
+
+type NestedDict = {
+  [key: string]: NestedDict | undefined
+}
+
+export const nestedFieldsToString = (nestedFields?: NestedDict) => {
+  if (!nestedFields) {
+    return ''
```

Comment:
```suggestion
    return '{ }'
```

return empty object?

Reply:
this generates any subfields which should go on to the graphql query, so either 
```
property
```
for a scalar field or
```
property {
  subproperty
}
```
for an object field, so if property has no sub properties you don't add anything, hence the empty string

Reply:
👍